desktop/cmp/persistenceManager/PersistenceManagerModel.ts

```diff
@@ -0,0 +1,400 @@
+    omitDefaultMenuComponent?: boolean;
+    /** Whether user can publish or edit globally shared objects. */
+    canManageGlobal: Thunkable<boolean>;
+    /** Async callback triggered when view changes. Should be used to recreate the affected models. */
+    onChangeAsync: (value: T) => void;
+    /** Used to persist this model's selected ID. */
+    persistWith: PersistOptions;
+    /** True (default) to render a save button alongside the primary menu button when dirty. */
+    enableTopLevelSaveButton?: boolean;
+    /** Fn to produce a new, empty object - can be async. */
+    newObjectFnAsync?: () => T;
+}
+
+export class PersistenceManagerModel<T extends PlainObject = PlainObject> extends HoistModel {
+    //------------------------
+    // Persistence Provider
+    // Pass this to models that implement `persistWith` to include their state in the view.
+    //------------------------
+    readonly provider: PersistOptions = {
+        getData: () => cloneDeep(this.pendingValue ?? this.value ?? {}),
+        setData: (value: T) => this.mergePendingValue(value)
+    };
+
+    readonly enableTopLevelSaveButton: boolean = true;
+
+    private readonly _canManageGlobal: Thunkable<boolean>;
+
+    readonly onChangeAsync?: (value: T) => void;
+
+    readonly entity: Entity;
+
+    readonly omitDefaultMenuComponent: boolean = false;
+
+    @observable.ref @managed manageDialogModel: ManageDialogModel;
+
+    @observable.ref @managed saveDialogModel: SaveDialogModel;
+    /** Current state of the active object, can include not-yet-persisted changes. */
+    @observable.ref pendingValue: T = null;
+
+    @observable.ref views: PersistenceView<T>[] = [];
+
+    @bindable selectedId: StoreRecordId;
+    @bindable favorites: string[] = [];
+
+    get canManageGlobal(): boolean {
+        return executeIfFunction(this._canManageGlobal);
+    }
+
+    get value(): T {
+        return this.selectedView?.value;
+    }
+
+    get selectedView(): PersistenceView<T> {
+        return this.views.find(it => it.id === this.selectedId);
+    }
+
+    @computed
+    get canSave(): boolean {
+        const {selectedView} = this;
+        return (
+            selectedView &&
+            this.isDirty &&
+            (this.canManageGlobal || !selectedView.isShared) &&
+            !this.loadModel.isPending
+        );
+    }
+
+    @computed
+    get isDirty(): boolean {
+        return !this.isEqualSkipAutosize(this.pendingValue, this.value);
+    }
+
+    get isShared(): boolean {
+        return !!this.selectedView?.isShared;
+    }
+
+    get favoritedViews(): PersistenceView<T>[] {
+        return this.views.filter(it => this.favorites.includes(it.token));
+    }
+
+    get viewTree(): TreeView[] {
+        const groupedViews = groupBy(this.views, 'group'),
+            sortedGroupKeys = keys(groupedViews).sort(),
+            ret = [];
+
+        if (this.favoritedViews.length > 0) {
```

Comment:
I wouldn't include the favorited views twice- the additional rendering of favorites with an accompanying divider feels more like a component-level concern.  Would just have this getter return the main tree.